src/opers.c

```diff
@@ -394,43 +365,43 @@ static Int IsSubsetFlagsCalls;
 /****************************************************************************
 **
 *F  IS_SUBSET_FLAGS( <flags1>, <flags2> ) . subset test with no safety check
+** checks is <flags2> is a subset of <flags1>
 */
 static Int IS_SUBSET_FLAGS(Obj flags1, Obj flags2)
 {
     Int    len1;
     Int    len2;
-    UInt * ptr1;
-    UInt * ptr2;
-    Int    i;
-
+    
+/* do the real work                                                    */
 #ifdef COUNT_OPERS
     IsSubsetFlagsCalls++;
 #endif
 
-    /* compare the bit lists                                               */
-    len1 = NRB_FLAGS(flags1);
-    len2 = NRB_FLAGS(flags2);
-    ptr1 = BLOCKS_FLAGS(flags1);
-    ptr2 = BLOCKS_FLAGS(flags2);
-    if (len1 < len2) {
-        for (i = len2 - 1; i >= len1; i--) {
-            if (ptr2[i] != 0)
-                return 0;
-        }
-        for (i = len1 - 1; i >= 0; i--) {
-            UInt x = ptr2[i];
-            if ((x & ptr1[i]) != x)
-                return 0;
-        }
-    }
-    else {
-        for (i = len2 - 1; i >= 0; i--) {
-            UInt x = ptr2[i];
-            if ((x & ptr1[i]) != x)
-                return 0;
+    len1 = SIZE_FLAGS(flags1);
+    len2 = SIZE_FLAGS(flags2);
+    if (len2 > len1)
+        return 0;
+    if (len2== 0)
+        return 1;
+    UInt i = 0;
+    UInt j = 0;
+    UInt2 x,y;
+    UInt2 *trues1, *trues2;
+    trues1 = (UInt2*)(CONST_ADDR_OBJ(flags1) + 3);
+    trues2 = (UInt2*)(CONST_ADDR_OBJ(flags2) + 3);
+    while (i < len1 && j < len2) {
+        x = trues1[i];
+        y = trues2[j];
+        if (x == y) {
+            i++;
+            j++;
+        } else if (y < x) {
+            return 0;
+        } else {
+            i++;
```

Comment:
So this keeps re-reading the same `trues2[j]` value -- probably fine, as it'll be in the L1 cache. And perhaps
the compiler will super optimize it anyway? But I still wonder how it compares to this:
```C
while (i < len1 && j < len2) {
  y = trues2[j++];
  do {
    x = trues1[i++];
  while (y > x);
  if (y < x)
    return 0;
}
```

Reply:
While this can wait, I'd try running this loop backwards -- flag lists often disagree at the end more than the beginning.

Reply:
That form might be better, but, as you say rereading from L1 doesn't seem to be much trouble. I hadn't seen the two loop version.